nexus-context-propagation/interceptor.go

```diff
@@ -0,0 +1,114 @@
+package nexuscontextpropagation
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"github.com/temporalio/samples-go/ctxpropagation"
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/interceptor"
+	"go.temporal.io/sdk/workflow"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type WorkerInterceptor struct {
+	interceptor.WorkerInterceptorBase
+	DataConverter converter.DataConverter
+}
+
+func (w *WorkerInterceptor) InterceptWorkflow(ctx workflow.Context, next interceptor.WorkflowInboundInterceptor) interceptor.WorkflowInboundInterceptor {
+	in := &workflowInboundInterceptor{parent: w}
+	in.Next = next
+	return in
+}
+
+func (w *WorkerInterceptor) InterceptNexusOperation(ctx context.Context, next interceptor.NexusOperationInboundInterceptor) interceptor.NexusOperationInboundInterceptor {
+	i := &nexusOperationInboundInterceptor{parent: w}
+	i.Next = next
+	return i
+}
+
+type workflowInboundInterceptor struct {
+	interceptor.WorkflowInboundInterceptorBase
+	parent *WorkerInterceptor
+}
+
+func (in *workflowInboundInterceptor) Init(next interceptor.WorkflowOutboundInterceptor) error {
+	out := &workflowOutboundInterceptor{parent: in.parent}
+	out.Next = next
+	return in.Next.Init(out)
+}
+
+type workflowOutboundInterceptor struct {
+	interceptor.WorkflowOutboundInterceptorBase
+	parent *WorkerInterceptor
+}
+
+type nexusErrorFuture struct {
+	workflow.Future
+}
+
+func newNexusErrorFuture(ctx workflow.Context, err error) nexusErrorFuture {
+	fut, settable := workflow.NewFuture(ctx)
+	settable.SetError(err)
+	return nexusErrorFuture{fut}
+}
+
+func (n nexusErrorFuture) GetNexusOperationExecution() workflow.Future {
+	// Return the same future
+	return n
+}
+
+// ExecuteNexusOperation implements interceptor.WorkflowOutboundInterceptor. It extracts values from workflow context
+// and propagates them via a Nexus header.
+func (out *workflowOutboundInterceptor) ExecuteNexusOperation(
+	ctx workflow.Context,
+	input interceptor.ExecuteNexusOperationInput,
+) workflow.NexusOperationFuture {
+	if values, ok := ctx.Value(ctxpropagation.PropagateKey).(ctxpropagation.Values); ok {
```

Comment:
In Java I can just copy the headers from the context directly because `MDC` only supports strings, this is fine though for Go